scipy/sparse/csgraph/tests/test_graph_laplacian.py

```diff
@@ -34,26 +45,32 @@ def _check_symmetric_graph_laplacian(mat, normed):
     else:
         sp_mat = sparse.csr_matrix(mat)
 
-    laplacian = csgraph.laplacian(mat, normed=normed)
+    mat_copy = np.copy(mat)
+    sp_mat_copy = sparse.csr_matrix(sp_mat, copy=True)
+
     n_nodes = mat.shape[0]
-    if not normed:
-        assert_array_almost_equal(laplacian.sum(axis=0), np.zeros(n_nodes))
-    assert_array_almost_equal(laplacian.T, laplacian)
-    assert_array_almost_equal(
-        laplacian, csgraph.laplacian(sp_mat, normed=normed).toarray())
+    explicit_laplacian = _explicit_laplacian(mat, normed=normed)
+    laplacian = csgraph.laplacian(mat, normed=normed, copy=copy)
+    sp_laplacian = csgraph.laplacian(sp_mat, normed=normed,
+                                     copy=copy)
+
+    if copy:
+        assert_array_almost_equal(mat, mat_copy)
+        _assert_allclose_sparse(sp_mat, sp_mat_copy)
+    else:
+        if not (normed and (np.issubdtype(mat.dtype, np.signedinteger)
+                            or np.issubdtype(mat.dtype, np.uint))):
+            assert_array_almost_equal(laplacian, mat)
+            if sp_mat.format == 'coo':
+                _assert_allclose_sparse(sp_laplacian, sp_mat)
 
-    assert_array_almost_equal(laplacian,
-            _explicit_laplacian(mat, normed=normed))
+    assert_array_almost_equal(laplacian, sp_laplacian.toarray())
```

Comment:
If we don't copy, we should still test that `mat != mat_copy` to show that it was really inline. (we can either check that we would raise an `AssertionError` with `assert_allclose`)

Reply:
I believe such a check would be redundant, since I already test that `laplacian == mat` and `laplacian` is never the same as `mat_copy`, which always implies `mat != mat_copy`